library/src/scala/Tuple.scala

```diff
@@ -33,6 +33,20 @@ sealed trait Tuple extends Any {
   inline def size[This >: this.type <: Tuple]: Size[This] =
     DynamicTuple.dynamicSize(this)
 
+  /** Given two tuples, `(A1, ..., An)` and `(B1, ..., Bn)`, returns a tuple
+   *  `((A1, B1), ..., (An, Bn))`. If the two tuples have different sizes,
+   *  the extra elements of the larger tuple will be disregarded.
+   */
+  inline def zip[This >: this.type <: Tuple, T2 <: Tuple](t2: T2): Zip[This, T2] =
```

Comment:
```
  /** Given two tuples, `(a1, ..., an)` and `(a1, ..., an)`, returns a tuple
   *  `((a1, b1), ..., (an, bn))`. If the two tuples have different sizes,
   *  the extra elements of the larger tuple will be disregarded.
   *  The result is typed as `((A1, B1), ..., (An, Bn))` if at least one of the tuple types has a `Unit` tail
   *  otherwise the result type is `(A1, B1) *: ... *: (Ai, Bi) *: Tuple`
   */
```